src/foundry/client/apps/sidebar/tab-base.d.ts

```diff
@@ -37,6 +37,10 @@ declare abstract class SidebarTab<
    */
   static override get defaultOptions(): ApplicationOptions;
 
+  override get id(): string;
+
+  override async getData(options?: Partial<Options> | undefined): MaybePromise<object>;
```

Comment:
I'm surprised this builds.

I tried to put together a code sample that does this and builds but it doesn't:
```ts
type MaybePromise<T> = T | Promise<T>;

class Foo {
    async foo(): MaybePromise<number> {
        return 1;
    };
}
```

This errors with
```
The return type of an async function or method must be the global Promise<T> type. Did you mean to write 'Promise<number>'?(1064)
```

So I'd expect you to have to get rid of the `async` modifier. Is this file not being included in the compilation for some reason or something? If so I feel like that's an issue that needs to be solved.

Reply:
Took my findings to #2443. Please remove `async` and check that this builds even if you remove `skipLibCheck` from the tsconfig. When it does it can get merged.

Reply:
Done; tab-base builds fine, the other issues are in other files. Should I merge #2443 into this branch then mark as dependent?